assets/js/index.js

```diff
@@ -0,0 +1,262 @@
+        <p>
+          UV index: <span class=${getUVIClassName(current.uvi)}>${
+    current.uvi
+  }</span>
+        </p>
+      </div>
+      <div class="current-city-img">
+        <img
+        src="https://openweathermap.org/img/w/${current.iconCode}.png"
+      />
+      </div>
+  </div>
+</div>`;
+
+  dashboardContainer.append(currentWeatherCard);
+};
+
+//construct forecast cards
+const renderForecastWeatherCards = (forecastData) => {
+  constructForecastCard = (each) => {
+    return `<div class="forecast-card">
+        <h5 class="forecast-card-title">${each.date}</h5>
+        <div class="forecast-card-body">
+            <div class="forecast-card-content">
+                <p class="card-text">Max Temp: ${each.max_temperature}&deg;F</p>
+                <p class="card-text">Min Temp: ${each.min_temperature}&deg;F</p>
+                <p class="card-text">Wind: ${each.wind} MPH</p>
+                <p class="card-text">Humidity: ${each.humidity}%</p>
+            </div>
+            <div class="forecast-card-img">
+                <img
+                src="https://openweathermap.org/img/w/${each.iconCode}.png"
+            />
+            </div>
+        </div>
+    </div>`;
+  };
+
+  const forecastCards = forecastData.map(constructForecastCard).join("");
+
+  const forecastCardsContainer = ` <div class="forecast-container">
+        <h3 class="forecast-title">7-Day Forecast:</h3>
+        <div class="forecast-card-container"> ${forecastCards}
+        </div>
+    </div>`;
+
+  dashboardContainer.append(forecastCardsContainer);
+};
+
+const renderWeatherCards = (weatherData) => {
+  if (weatherData) {
+    renderCurrentWeatherCard(weatherData.current);
+
+    renderForecastWeatherCards(weatherData.forecast);
+  } else {
+    renderAlertCard();
+  }
+};
+
+const renderRecentCities = () => {
+  //get cities from LS
+  const cities = JSON.parse(localStorage.getItem("recentCities")) ?? [];
+
+  cityList.empty();
+
+  const constructAndAppendCity = (city) => {
+    //construct each li item
+    const liEl = `<li data-city=${city} class="list-group-item">${city}</li>`;
+
+    //append to the list
+    cityList.append(liEl);
+  };
+
+  const handleClick = (event) => {
+    const target = $(event.target);
+
+    // if click is from li
+    if (target.is("li")) {
+      //get city data-name
+      const cityDataName = target.data("city");
+
+      //render weather inf with cityDataName
+      renderWeatherInfo(cityDataName);
+    }
+  };
+
+  cityList.on("click", handleClick);
+
+  cities.forEach(constructAndAppendCity);
+};
+
+const renderWeatherInfo = async (cityName) => {
+  //get data from API
+  const weatherData = await getWeatherData(cityName);
+
+  //remove the last city forecast
+  dashboardContainer.empty();
+
+  renderWeatherCards(weatherData);
+};
+
+const handleSearch = async (event) => {
+  event.preventDefault();
+
+  const cityName = $("#city-input").val();
+
+  if (cityName) {
+    renderWeatherInfo(cityName);
```

Comment:
This is the line which is causing your LS not to be updated properly on search, meaning the recently searched item only gets added to the recent cities list on a refresh. Since `renderWeatherInfo` is an `async` function we need to `await` it when we call it (so here it would be `await renderWeatherInfo(cityName)`. 

As we are not awaiting it at the moment, we are not waiting for this function to finish running before we move on to the next line `renderRecentCities()`. For this reason, the city has not actually been added to local storage when line 230 is run and so it doesn't appear straight away. If you add some `console.log`s to your code in each of the functions, you will see that they run in an unexpected order.

Reply:
thanks for this nice explanation, got your point. Fixed!!!!